earth2grid/spatial.py

```diff
@@ -44,3 +44,10 @@ def ang2vec(lon, lat):
     y = torch.cos(lat) * torch.sin(lon)
     z = torch.sin(lat)
     return (x, y, z)
+
+
+def vec2ang(x, y, z):
+    """convert lon,lat in radians to cartesian coordinates"""
+    lat = torch.asin(z)
```

Comment:
This assumes your inputs are already normalized. If you want it to work with non-normalized inputs, you would need
```suggestion
    lat = torch.atan2(z, torch.hypot(y, x))
```